source/Irrlicht/CBAWMeshWriter.cpp

```diff
@@ -159,34 +211,51 @@ namespace irr {
 		{
 			_ctx.headers.clear();
 
+			bool isMeshAnimated = true;
+			ICPUSkinnedMesh* skinnedMesh = 0;
+
 			if (_mesh)
 			{
-				core::BlobHeader bh;
+				skinnedMesh = dynamic_cast<ICPUSkinnedMesh*>(_mesh);
```

Comment:
can be a static_cast or reinterpret_cast

Reply:
Nope, `dynamic_cast` is the only one really safe here. `reinterpret_cast` just gave me crash in analogic situation (base class can have a bit different offset and only `dynamic_cast` cares about that).

Reply:
ICPUSkinnedMesh is a direct inheritor of ICPUMesh, so reinterpret_cast should be safe.

if you were casting from IReferenceCounted, then you'd have a problem.

Reply:
'fixed'